server/src/main/webapp/WEB-INF/rails/webpack/models/base/cache.ts

```diff
@@ -88,10 +88,14 @@ export abstract class AbstractObjCache<T> implements ObjectCache<T> {
   }
 
   invalidate() {
-    this.empty();
+    this.markStale();
     this.primed = false;
   }
 
+  protected markStale() {
```

Comment:
Adding another abstraction only because I like the idea of still allowing an explicit `empty()` if ever needed. The `data` field is marked `private` and would be otherwise inaccessible in typescript.